R/filter-helpers.R

```diff
@@ -13,19 +13,20 @@
 #'   dm_nrow()
 #' @export
 dm_nrow <- function(dm) {
+  check_not_zoomed(dm)
   # FIXME: with "direct" filter maybe no check necessary: but do we want to issue
   # a message in case the filters haven't been applied yet?
   check_no_filter(dm)
-  map_dbl(dm_get_tables(dm), ~ as.numeric(pull(collect(count(.)))))
+  map_int(dm_get_tables_impl(dm), ~ as.integer(pull(collect(count(.)))))
```

Comment:
Protects against > 2³¹ rows.

```suggestion
  map_dbl(dm_get_tables_impl(dm), ~ as.numeric(pull(collect(count(.)))))
```